controllers/operator/common_controller_test.go

```diff
@@ -265,9 +265,124 @@ func TestReadSubjectNoCertificate(t *testing.T) {
 	assertSubjectFromFileFails(t, "testdata/certificates/just_key")
 }
 
+func TestFailWhenRoleAndRoleRefsAreConfigured(t *testing.T) {
+	ctx := context.Background()
+	customRole := mdbv1.MongoDBRole{
+		Role:                       "foo",
+		AuthenticationRestrictions: []mdbv1.AuthenticationRestriction{},
+		Db:                         "admin",
+		Roles: []mdbv1.InheritedRole{{
+			Db:   "admin",
+			Role: "readWriteAnyDatabase",
+		}},
+	}
+	roleResource := DefaultClusterMongoDBRoleBuilder().Build()
+	roleRef := mdbv1.MongoDBRoleRef{
+		Name: roleResource.Name,
+		Kind: util.ClusterMongoDBRoleKind,
+	}
+	assert.Nil(t, customRole.Privileges)
+	rs := mdbv1.NewDefaultReplicaSetBuilder().SetRoles([]mdbv1.MongoDBRole{customRole}).SetRoleRefs([]mdbv1.MongoDBRoleRef{roleRef}).Build()
+
+	kubeClient, omConnectionFactory := mock.NewDefaultFakeClient()
+	controller := NewReconcileCommonController(ctx, kubeClient)
+	mockOm, _ := prepareConnection(ctx, controller, omConnectionFactory.GetConnectionFunc, t)
+
+	result := controller.ensureRoles(ctx, rs.Spec.Security.Roles, rs.Spec.Security.RoleRefs, mockOm, kube.ObjectKeyFromApiObject(rs), &zap.SugaredLogger{})
+	assert.False(t, result.IsOK())
+	assert.Equal(t, status.PhaseFailed, result.Phase())
```

Comment:
Nit: Are we expecting failure here because of validations error ?
If yes we may want to make it explicit, e.g testing the err message